internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -616,249 +616,137 @@ func (b *BeatsMonitor) injectMetricsInput(
 		failureThreshold = &defaultValue
 	}
 	monitoringNamespace := b.monitoringNamespace()
-	fixedAgentName := strings.ReplaceAll(agentName, "-", "_")
+
+	//create a new map with the monitoring beats included
+	componentListWithMonitoring := map[string]string{
+		fmt.Sprintf("beat/%s", monitoringMetricsUnitID): "metricbeat",
+		fmt.Sprintf("http/%s", monitoringMetricsUnitID): "metricbeat",
+		monitoringFilesUnitsID:                          "filebeat",
+	}
+	for k, v := range componentIDToBinary {
+		componentListWithMonitoring[k] = v
+	}
+
 	// beatStreams and streams MUST be []interface{} even if in reality they are []map[string]interface{}:
```

Comment:
Probably should update this comment to match the code changes below.

Reply:
I'll move this comment to the functions themselves, since you don't need to know the slice type here.